### PR TITLE
Add support for order promotion API

### PR DESCRIPTION
**Pull request by poulch**
This PR require API to be on staging before merge.

## What type of PR is this?
- [ ] 💅 Refactor
- [x] 🌟 Feature
- [ ] 🔥 Bug Fix
- [ ] 🔩 Maintenance
- [ ] 🛠 Workflow CI/CD changes

## Related Issues or Documents

- closes https://github.com/saleor/saleor-dashboard/issues/4645
- closes https://github.com/saleor/saleor-dashboard/issues/4649
- closes https://github.com/saleor/saleor-dashboard/issues/4665

## Have you written tests?
- [x] Yes!
- [ ] No... here is why: _Writing tests are mandatory, please replace this text with why test are not included in this PR_

### CODE REVIEW

File: src/discounts/components/DiscountRules/componenets/RulesList/components/RuleLabel/RuleLabel.tsx [hunk 1]
```diff
@@ -8,6 +10,7 @@ interface RuleLabelProps {
 
 export const RuleLabel = ({ ruleName }: RuleLabelProps) => {
```
Comment (andrzejewsky) [inline src/discounts/components/DiscountRules/componenets/RulesList/components/RuleLabel/RuleLabel.tsx:11]: does it have to be a component if it returns just a text?

File: src/discounts/models/OrderRule/prepareConditions.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    .map(([id, value]) => {
+      if (id === "OR") {
+        return prepareOrderConditions(
+          (value as any[]).reduce(
+            (acc: OrderPredicateAPI, val: Record<string, unknown>) => {
+              acc = { ...acc, ...val };
+              return acc;
+            },
+            {} as OrderPredicateAPI,
+          ),
```
Comment (andrzejewsky) [inline src/discounts/models/OrderRule/prepareConditions.ts:24]: **suggestion**: make function for it where the name indicates what it reduces

File: src/discounts/components/DiscountRules/componenets/RuleForm/components/RuleConditionName/useConditionNames.ts [hunk 3]
```diff
@@ -1,17 +1,48 @@
...
-  const [conditionNames] = useState<Option[]>(getCatalogConditionOptions(intl));
+  const [conditionNames, setConditionNames] = useState<Option[]>([]);
+
+  useEffect(() => {
+    if (discountType === PromotionTypeEnum.CATALOGUE) {
+      setConditionNames(getCatalogConditionOptions(intl));
+    } else {
+      setConditionNames(getOrderConditionOptions(intl));
+    }
+  }, [discountType]);
```
Comment (andrzejewsky) [inline src/discounts/components/DiscountRules/componenets/RuleForm/components/RuleConditionName/useConditionNames.ts:17]: this sounds like separated context that can be within separated hook

**Review by andrzejewsky — APPROVED** [review]

**Review by andrzejewsky — APPROVED** [review]

File: src/discounts/models/helpers.test.ts [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+            ],
+          },
+        },
+      ],
+    } as CataloguePredicateAPI;
+
+    expect(hasPredicateNestedConditions(catalogPredicate)).toBe(true);
+  });
+
+  it("hould return true if predicate has nested AND property", () => {
```
Comment (yellowee) [inline src/discounts/models/helpers.test.ts:115]:
```suggestion
  it("should return true if predicate has nested AND property", () => {
```

**Review by yellowee — DISMISSED** [review]

**Review by yellowee — APPROVED** [review]

File: .changeset/thirty-rocks-glow.md
Comment (Droniu) [inline .changeset/thirty-rocks-glow.md:1]: should we have two changesets?

File: src/discounts/queries.ts [hunk 5]
```diff
@@ -201,3 +201,50 @@ export const ruleConditionsSelectedOptionsDetails = gql`
...
+          hasOrders
+          orderSettings {
+            markAsPaidStrategy
+            deleteExpiredOrdersAfter
+            allowUnpaidOrders
+          }
+          paymentSettings {
+            defaultTransactionFlowStrategy
+          }
+        }
```
Comment (Droniu) [inline src/discounts/queries.ts:240]: **suggestion** we could use fragments for some of these

**Review by Droniu — APPROVED** [review]

**Pull request merged**